pydantic/_internal/_generate_schema.py

```diff
@@ -522,6 +524,12 @@ def _typed_dict_schema(
         Hence to avoid creating validators that do not do what users expect we only
         support typing.TypedDict on Python >= 3.11 or typing_extension.TypedDict on all versions
         """
+        if origin is not None:
+            typeddict_typevars_map = dict(zip(origin.__parameters__, typed_dict_cls.__args__))
+            typed_dict_cls = origin
+        else:
+            typeddict_typevars_map = {}
```

Comment:
this is a little hacky where I replace the typed_dict_cls with the origin, but it made for a small diff here and I think it's clear enough. (What's happening is I'm pulling the typevars mapping off of the origin+_GenericAlias pair, and then applying that mapping down below to each field in turn)

Reply:
Makes sense. I'm not sure how we handle it for BaseModel / dataclasses (the latter I think the answer is we don't, yet?) but we should probably make them all mirror images of each other?

Reply:
I agree — I'll just note that it won't surprise me at all if this needs (simplicity-reducing) changes to work in the most general cases (like recursive nested generics that involve chains of generic basemodels, dataclasses, and typeddicts sharing parameters etc.).

But considering this handles what are likely to be by far the most common cases, I think it's worth doing this now even if we never perfectly handle the most complicated things imaginable.

I do plan to do another unifying pass over the generics stuff at some point though, just to be clear.